Index: mt7603e_wifi_4.0.0.5_20140511/mt_wifi/os/linux/rt_linux.c
===================================================================
--- mt7603e_wifi_4.0.0.5_20140511.orig/mt_wifi/os/linux/rt_linux.c
+++ mt7603e_wifi_4.0.0.5_20140511/mt_wifi/os/linux/rt_linux.c
@@ -43,8 +43,8 @@
 #endif
 
 #if defined(CONFIG_RA_HW_NAT) || defined(CONFIG_RA_HW_NAT_MODULE)
-#include "../../../../../../net/nat/hw_nat/ra_nat.h"
-#include "../../../../../../net/nat/hw_nat/frame_engine.h"
+#include "../../../linux-2.6.36/net/nat/hw_nat/ra_nat.h"
+#include "../../../linux-2.6.36/net/nat/hw_nat/frame_engine.h"
 #endif
 
 /* TODO */
Index: mt7603e_wifi_4.0.0.5_20140511/mt_wifi/os/linux/rt_profile.c
===================================================================
--- mt7603e_wifi_4.0.0.5_20140511.orig/mt_wifi/os/linux/rt_profile.c
+++ mt7603e_wifi_4.0.0.5_20140511/mt_wifi/os/linux/rt_profile.c
@@ -30,8 +30,8 @@
 #endif
 
 #if defined (CONFIG_RA_HW_NAT)  || defined (CONFIG_RA_HW_NAT_MODULE)
-#include "../../../../../../net/nat/hw_nat/ra_nat.h"
-#include "../../../../../../net/nat/hw_nat/frame_engine.h"
+#include "../../../linux-2.6.36/net/nat/hw_nat/ra_nat.h"
+#include "../../../linux-2.6.36/net/nat/hw_nat/frame_engine.h"
 #endif
 
 
