Index: mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/cmm_cfg.c
===================================================================
--- mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103.orig/rlt_wifi/common/cmm_cfg.c	2014-01-03 15:53:38.000000000 +0800
+++ mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/cmm_cfg.c	2014-03-07 16:59:12.184166962 +0800
@@ -1245,6 +1245,7 @@
 		{
 			RT_CMD_IOCTL_RATE *pRate = (RT_CMD_IOCTL_RATE *)pData;
 			HTTRANSMIT_SETTING HtPhyMode;
+			UINT8 BW = 0, GI = 0, MCS = 0;
 
 #ifdef APCLI_SUPPORT
 			if (pRate->priv_flags == INT_APCLI)
@@ -1263,8 +1264,21 @@
 				MBSS_PHY_MODE_RESET(pObj->ioctl_if, HtPhyMode);
 #endif /* MBSS_SUPPORT */
 			}
-			RtmpDrvMaxRateGet(pAd, HtPhyMode.field.MODE, HtPhyMode.field.ShortGI,
-							HtPhyMode.field.BW, HtPhyMode.field.MCS,
+
+#ifdef DOT11_VHT_AC
+           if (HtPhyMode.field.BW == BW_40 && pAd->CommonCfg.vht_bw == VHT_BW_80) {
+               BW = 2;
+               GI = pAd->CommonCfg.vht_sgi_80;
+           }
+           else
+#endif /* DOT11_VHT_AC */
+           {
+               BW = HtPhyMode.field.BW;
+               GI = HtPhyMode.field.ShortGI;
+           }
+
+           RtmpDrvMaxRateGet(pAd, HtPhyMode.field.MODE, GI,
+                           BW, HtPhyMode.field.MCS,
 							pAd->Antenna.field.TxPath,
 							(UINT32 *)&pRate->BitRate);
 		}
Index: mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/rt_os_util.c
===================================================================
--- mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103.orig/rlt_wifi/common/rt_os_util.c	2014-01-03 15:53:38.000000000 +0800
+++ mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/rt_os_util.c	2014-03-07 17:00:30.612165984 +0800
@@ -23,21 +23,21 @@
 
 
  UINT32 RalinkRate[256] =
-        {2,  4, 11, 22, 12, 18, 24, 36, 48, 72, 96, 108, 109, 110, 111, 112,/* CCK and OFDM */
-        13, 26,   39,  52,  78, 104, 117, 130, 26,  52,  78, 104, 156, 208, 234, 260,
-        39, 78,  117, 156, 234, 312, 351, 390, /* BW 20, 800ns GI, MCS 0~23 */
-        27, 54,   81, 108, 162, 216, 243, 270, 54, 108, 162, 216, 324, 432, 486, 540,
-        81, 162, 243, 324, 486, 648, 729, 810, /* BW 40, 800ns GI, MCS 0~23 */
-        14, 29,   43,  57,  87, 115, 130, 144, 29, 59,   87, 115, 173, 230, 260, 288,
-        43, 87,  130, 173, 260, 317, 390, 433, /* BW 20, 400ns GI, MCS 0~23 */
-        30, 60,   90, 120, 180, 240, 270, 300, 60, 120, 180, 240, 360, 480, 540, 600,
-        90, 180, 270, 360, 540, 720, 810, 900, /* BW 40, 400ns GI, MCS 0~23 */
-        13, 26,   39,  52,  78, 104, 117, 130, 156, /* 11ac: 20Mhz, 800ns GI, MCS: 0~8 */
-        27, 54,   81, 108, 162, 216, 243, 270, 324, 360, /*11ac: 40Mhz, 800ns GI, MCS: 0~9 */
-        59, 117, 176, 234, 351, 468, 527, 585, 702, 780, /*11ac: 80Mhz, 800ns GI, MCS: 0~9 */
-        14, 29,   43,  57,  87, 115, 130, 144, 173, /* 11ac: 20Mhz, 400ns GI, MCS: 0~8 */
-        30, 60,   90, 120, 180, 240, 270, 300, 360, 400, /*11ac: 40Mhz, 400ns GI, MCS: 0~9 */
-        65, 130, 195, 260, 390, 520, 585, 650, 780, 867, /*11ac: 80Mhz, 400ns GI, MCS: 0~9 */
+        {2,  4, 11, 22, 12, 18, 24, 36, 48, 72, 96, 108, 109, 110, 111, 112,/* CCK and OFDM */         /* 0 ~ 15 */
+        13, 26,   39,  52,  78, 104, 117, 130, 26,  52,  78, 104, 156, 208, 234, 260,                  /* 16 ~ 31 */
+        39, 78,  117, 156, 234, 312, 351, 390, /* BW 20, 800ns GI, MCS 0~23 */                     /* 32 ~ 39 */
+        27, 54,   81, 108, 162, 216, 243, 270, 54, 108, 162, 216, 324, 432, 486, 540,              /* 40 ~ 55 */
+        81, 162, 243, 324, 486, 648, 729, 810, /* BW 40, 800ns GI, MCS 0~23 */                     /* 56 ~ 63 */
+        14, 29,   43,  57,  87, 115, 130, 144, 29, 59,   87, 115, 173, 230, 260, 288,                  /* 64 ~ 79 */
+        43, 87,  130, 173, 260, 317, 390, 433, /* BW 20, 400ns GI, MCS 0~23 */                     /* 80 ~ 87 */
+        30, 60,   90, 120, 180, 240, 270, 300, 60, 120, 180, 240, 360, 480, 540, 600,              /* 88 ~ 103 */
+        90, 180, 270, 360, 540, 720, 810, 900, /* BW 40, 400ns GI, MCS 0~23 */                     /* 104 ~ 111 */
+        13, 26,   39,  52,  78, 104, 117, 130, 156, /* 11ac: 20Mhz, 800ns GI, MCS: 0~8 */          /* 112 ~ 120 */
+        27, 54,   81, 108, 162, 216, 243, 270, 324, 360, /*11ac: 40Mhz, 800ns GI, MCS: 0~9 */      /* 121 ~ 130 */
+        59, 117, 176, 234, 351, 468, 527, 585, 702, 780, /*11ac: 80Mhz, 800ns GI, MCS: 0~9 */      /* 131 ~ 140 */
+        14, 29,   43,  57,  87, 115, 130, 144, 173, /* 11ac: 20Mhz, 400ns GI, MCS: 0~8 */          /* 141 ~ 149 */
+        30, 60,   90, 120, 180, 240, 270, 300, 360, 400, /*11ac: 40Mhz, 400ns GI, MCS: 0~9 */      /* 150 ~ 159 */
+        65, 130, 195, 260, 390, 520, 585, 650, 780, 867, /*11ac: 80Mhz, 400ns GI, MCS: 0~9 */      /* 160 ~ 169 */
         0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16,17,18,19,
         20,21,22,23,24,25,26,27,28,29,30,31,32,33,34,35,36,37,38,39,
         40,41,42,43,44,45,46,47}; /* 3*3 */
@@ -58,6 +58,8 @@
        if (MODE >= MODE_VHT)
        {
                if (BW == 0/*20Mhz*/) {
+                        if (MCS > 8)
+                                MCS = 8;
                        rate_index = 112 +
                        ((UCHAR)ShortGI * 29) +
                        ((UCHAR)MCS);
Index: mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/include/ap_mbss.h
===================================================================
--- mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103.orig/rlt_wifi/include/ap_mbss.h	2014-01-03 15:53:37.000000000 +0800
+++ mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/include/ap_mbss.h	2014-03-07 16:59:12.184166962 +0800
@@ -61,7 +61,6 @@
                 {       \
                         __HtPhyMode.field.MODE = MODE_VHT;                                      \
                         __HtPhyMode.field.MCS = 9;                                                      \
-                        __HtPhyMode.field.BW = BW_80;   \
                }       \
 	}
 
Index: mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/cmm_info.c
===================================================================
--- mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103.orig/rlt_wifi/common/cmm_info.c	2014-01-03 15:53:38.000000000 +0800
+++ mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/common/cmm_info.c	2014-03-07 16:59:12.188166962 +0800
@@ -4900,11 +4900,10 @@
 			&& (pEntry->Sst == SST_ASSOC))
 		{
 			DataRate=0;
-			//getRate(pEntry->HTPhyMode, &DataRate);
 			RtmpDrvMaxRateGet(pAd, pEntry->HTPhyMode.field.MODE, pEntry->HTPhyMode.field.ShortGI,
-		                          pEntry->HTPhyMode.field.BW, pEntry->HTPhyMode.field.MCS,
-		                          (pEntry->MaxHTPhyMode.field.MCS>>4)+1, &DataRate);
-			DataRate /= 500000;
+		                          pEntry->HTPhyMode.field.BW, (pEntry->HTPhyMode.field.MCS & 0xf),
+		                          (pEntry->HTPhyMode.field.MCS>>4)+1, &DataRate);
+			DataRate /= 1000000; /* Bit/s to Mbps*/;
 			
 			printk("%02X:%02X:%02X:%02X:%02X:%02X  ", PRINT_MAC(pEntry->Addr));
 			printk("%-4d", (int)pEntry->Aid);
@@ -4926,7 +4925,7 @@
 			printk("%-6s", get_bw_str(pEntry->HTPhyMode.field.BW));
 #ifdef DOT11_VHT_AC
 			if (pEntry->HTPhyMode.field.MODE == MODE_VHT)
-				printk("%dS-M%d", ((pEntry->HTPhyMode.field.MCS>>4) + 1), (pEntry->HTPhyMode.field.MCS & 0xf));
+				printk("%dS-M%-2d", ((pEntry->HTPhyMode.field.MCS>>4) + 1), (pEntry->HTPhyMode.field.MCS & 0xf));
 			else
 #endif /* DOT11_VHT_AC */
 			printk("%-6d", pEntry->HTPhyMode.field.MCS);
