Index: mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/ap/ap.c
===================================================================
--- mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103.orig/rlt_wifi/ap/ap.c
+++ mt7612e_mt7602e_LinuxAP_3.0.2.9_iPA_20140103/rlt_wifi/ap/ap.c
@@ -2242,9 +2242,12 @@ VOID APOverlappingBSSScan(RTMP_ADAPTER *
 		pAd->CommonCfg.AddHTInfo.AddHtInfo.ExtChanOffset = 0;
 		pAd->CommonCfg.LastBSSCoexist2040.field.BSS20WidthReq = 1;
 		pAd->CommonCfg.Bss2040CoexistFlag |= BSS_2040_COEXIST_INFO_SYNC;
+ 		//FIXME: Fix HT20+40 Coexistence, with intel 6205agn 1Mbps problem.
+ 		pAd->CommonCfg.Bss2040NeedFallBack = 1;
+ 		pAd->CommonCfg.RegTransmitSetting.field.EXTCHA = 0;
 	}
 
-	return;	
+	return;
 }
 #endif /* DOT11N_DRAFT3 */
 #endif /* DOT11_N_SUPPORT */
